modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/OpenAPIYamlGenerator.java

```diff
@@ -61,6 +75,44 @@ public void processOpts() {
         }
         LOGGER.info("Output file [outputFile={}]", outputFile);
         supportingFiles.add(new SupportingFile("openapi.mustache", outputFile));
+
+        addMustacheLambdas(additionalProperties);
+    }
+
+    private void addMustacheLambdas(Map<String, Object> objs) {
```

Comment:
Since the same method is used in `AbstractCSharpCodegen`, `AbstractFSharpCodegen` and `KotlinServerCodegen` I think is would make sense to mutualize it move it somewhere (instead of all those copy-pasted methods)

Maybe as protected method in `DefaultCodegen`
Or as public static method in a utility class.

@MichalFoksa What do you think?
@jimschubert any opinion on this?

Reply:
Will look into it later

Reply:
@jmini I agree. I didn't put it in the DefaultCodegen type because I wasn't sure if would be acceptable globally. If moved there, we'll want to document these as available Mustache lambdas in our template docs.

Reply:
How would you like it? As a new method in CodegenConfig interface and 

1. an empty method in DefaultCodegen which gets called by DefaultGenerator and each generator uses it to register own lambdas. 
    `DefaultCodegen.addMustacheLambdas() {}`


2. Or do you want it to already register some lambdas, for exapmle:
```
    DefaultCodegen.addMustacheLambdas() {
        lambdas = new ImmutableMap.Builder<String, Mustache.Lambda>()
                .put("lowercase", new LowercaseLambda().generator(this))
                .put("uppercase", new UppercaseLambda())
                .put("titlecase", new TitlecaseLambda())
                .put("camelcase", new CamelCaseLambda().generator(this))
                .put("indented", new IndentedLambda())
                .....
                .build();
        objs.put("lambda", lambdas);
    }
```
?

Reply:
@MichalFoksa: I do not think that we need the method in the `CodegenConfig` interface. If I understand @jimschubert correctly the idea is: `2.` just register the Lambdas that we have for all generators. And add some explanations about usage in [`templating.md`](https://github.com/OpenAPITools/openapi-generator/blob/master/docs/templating.md) probably as new section "Lambdas" after "Variables" and before "Extensions".

Can you create a follow up PR?

Reply:
It _might_ need to be on the interface as well, as I believe another generator (C++ maybe) created lambdas with different names. I'll take a look later when I have a chance.

Reply:
I will create a follow up PR. In meantime I have discovered #3364.

Reply:
Here you are the follow up PR #3368